linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/VersionSubstitutingModelResolver.java

```diff
@@ -68,16 +68,16 @@
   }
 
   /**
-   * Returns model source for {@code groupId:artifactId:version}. The version is substituted if
-   * {@link #versionSubstitution} contains the versionless coordinates ({@code groupId:artifactId})
-   * in its keys.
+   * Returns a file model source for {@code groupId:artifactId:version}.
+   * The version is substituted if {@link #versionSubstitution} contains the
+   * versionless coordinates ({@code groupId:artifactId}) in its keys.
    */
   @Override
-  public ModelSource resolveModel(String groupId, String artifactId, String version)
+  public ModelSource2 resolveModel(String groupId, String artifactId, String version)
```

Comment:
I feel weird that the function return type is different from the super class.  http://maven.apache.org/ref/3.6.3/maven-model-builder/apidocs/org/apache/maven/model/resolution/ModelResolver.html#resolveModel-java.lang.String-java.lang.String-java.lang.String-

```
ModelSource resolveModel(String groupId,
                         String artifactId,
                         String version)
                  throws UnresolvableModelException
```

Shouldn't this be fixed at `org.apache.maven.model.resolution.ModelResolver`?

Reply:
Possibly, though it might need to be that way for backwards compatibility.

This is a covariant return type introduced in Java 5. It still fulfills the contract of the superclass/superinterface .

Reply:
This case was return value type, which is not used in method lookup. But what if there’s an argument of ModelSource.

Reply:
> covariant

Right. That’s the difference from the case of the arguments. Thank you.